pandas/core/indexes/base.py

```diff
@@ -251,6 +252,10 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
         if name is None and hasattr(data, 'name'):
             name = data.name
 
+        if name is not None and not is_hashable(name):
+            raise TypeError(cls.__name__ +
```

Comment:
can you use ``.format()`` here instead (and whenever you are composing an error string)